Source/WEB_ROOT/Inventory/js/editAsset.js

```diff
@@ -0,0 +1,206 @@
+	 
+	if (document.getElementById('isStudentCheckBox').checked) 
+	{
+		assignedType = 1;
+	} else {
+		assignedType = 0;
+	};
+
+	var url = "scripts/AssignedToDropDown.html?type=" + assignedType +"&id=" + "&schoolid=" + $j('#schoolDropDown :selected').val();
+	//console.log(url);
+	$j.ajax(
+	{
+		url: url, 
+		dataType: "text",
+		success: function(result)
+		{
+			
+			if (assignedType == 1) 
+			{
+				//push students to the dropdown
+				var options = '<select id="studentsToDropdown">' + result + '</select>';
+				document.getElementById( 'studentsToDropdown' ).outerHTML = options;
+			} else {
+				//push teachers to the dropdown
+				var options = '<select id="teachersDropdown">' + result + '</select>';
+				document.getElementById( 'teachersDropdown' ).outerHTML = options;
+			};
+			createlisteners();
+		}
+	});
+}
+
+function checkForDup(assetid)
+{
+    var serial = document.getElementById("serialInput").value
+
+    $j.ajax(
+	{
+		url: "scripts/checkDupSerial.html?serial=" + serial + "&assetid=" + assetid, 
+		dataType: "text",
+		success: function(result)
+		{
+            var warning = document.getElementById("serialDupWarning");
+            
+			if(result.length > 0)
+			{
+				warning.style.display = "block";
+			}else
+			{
+				warning.style.display = "none";
+			}
+		}
+	});
+}
+
+function toggleIsStudent()
+{
+    //console.log(document.getElementById('isStudentCheckBox').checked);
+    //I need to verify that true = 1 and false = 0 or null in the database
+  if (document.getElementById('isStudentCheckBox').checked) 
+  {
+	  document.getElementById('isStudentCheckBoxText').innerHTML = "<font color=green>Student - </font>";
+	  document.getElementById('isStudent_dcid').value = "1";
+      showHideFormElements("studentDropDownElement", true);
+      showHideFormElements("teacherDropDownElement", false);
+  } else {
+	  document.getElementById('isStudentCheckBoxText').innerHTML = "<font color=blue>Teacher - </font>";
+	  document.getElementById('isStudent_dcid').value = "0";
+      showHideFormElements("studentDropDownElement", false);
+      showHideFormElements("teacherDropDownElement", true);
+  }
+  SetDropDowns();
+}
+
+// show or hide all the form elements
+    function showHideFormElements(className, value)
+    {
+        var elements = document.getElementsByClassName(className);
+        //console.log(elements);
+        for (var i = 0; i < elements.length; i++)
+        {
+            if(value === true)
+            {
+                elements[i].style.display = "block";
+            }else{
+                elements[i].style.display = "none";
+            }
+        }
+    }
+// End Show / Hide Elements
+
+// Asset Type Dropdown section
+    $j('#commontypes').change(function(){
+            TypeChanged();
+        });
+
+    function TypeChanged()
+    {
+        document.getElementById("storetype").value = $j('#commontypes :selected').val();
+    }
+// End
+
+// One time page load checks
+function init(assetid)

```

Comment:
I would likely move this to the top of the class, it looks like a core method. If you get a chance check out "Clean Code" by Robert Martin. He'll give you this idea that your big meaty methods that really define the work on a macro level come first in the class. Giving the reader a sense of the classes purpose. The smaller methods that assist those large methods come further down.